blocks/core/style-overrides/heading.php

```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * Registers the heading block styles.
+ *
+ * @since 1.3.0
+ */
+function register_heading_block_styles() {
```

Comment:
if we are using functional programming, than we shouldn't have function names that could easily by accident be created in other locations.
If you are wanting to do this exact same pattern for all style overrides, and not use OOP, than maybe consider anonymous functions in your hook